code-for-canada/project-thundercat
# [User Model] Saving the missing registration fields in the database

# Description

**Important:**

===============================================================

Note that may need to manually migrate these migrations in your local DB, since we're adding mandatory fields in an already populated table (custom_models_user table). In order to do that, login to the backend container and execute the following commands:

1. _python manage.py makemigrations custom_models_
2. _select option 1 by entering "1"_
3. _timezone.now_
3. _python manage.py migrate_

===============================================================

This PR is introducing an updated user model that allows us to save all the registration fields in the database. Here is the list of the fields that get saved in the database:

- First Name
- Last Name
- Date of Birth (format: DD/MM/---Y)
- Email (including Username)
- PRI or Military Number (if provided)
- Password
- ID (auto generated)

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

**Filled Registration Form:**

![image](https://user-images.githubusercontent.com/23021242/59127879-a7ebd280-8936-11e9-895c-b25996177660.png)

**Results (JSON Version):**

![image](https://user-images.githubusercontent.com/23021242/59127931-d4075380-8936-11e9-9562-7755bb5b95f8.png)

**Results (DB Version):**

![image](https://user-images.githubusercontent.com/23021242/59127989-fa2cf380-8936-11e9-92c2-0eb74ba6927d.png)

# Testing

Manual steps to reproduce this functionality:

1.  Create a new user.
2.  Verify either in the DB or using the user API (http://localhost/api/auth/users/) that you get the right data.

# Checklist

Applicable for all code changes.
